starknet_devnet/blueprints/rpc/utils.py

```diff
@@ -47,15 +51,30 @@ def get_block_by_block_id(block_id: BlockId) -> dict:
         raise RpcError(code=24, message="Invalid block id") from ex
 
 
-def assert_block_id_is_latest(block_id: BlockId) -> None:
+def assert_block_id_is_latest_or_pending(block_id: BlockId) -> None:
     """
-    Assert block_id is "latest" and throw RpcError otherwise
+    Assert block_id is "latest"/"pending" or a block hash or number of "latest"/"pending" block and throw RpcError otherwise
     """
-    if block_id != "latest":
-        raise RpcError(
-            code=-1,
-            message="Calls with block_id != 'latest' are not supported currently.",
-        )
+    last_block = state.starknet_wrapper.blocks.get_last_block()
```

Comment:
This last block retrieval can be moved inside of the `if isinstance(block_id, dict)` clause